src/neps/optimizers/bayesian_optimization/acquisition_functions/mf_ei.py

```diff
@@ -46,13 +103,83 @@ def preprocess(self, x: Iterable) -> Tuple[Iterable, Iterable]:
             inc_list.append(inc)
 
         return x, torch.Tensor(inc_list)
+    
+    def preprocess_gp(self, x: Iterable) -> Tuple[Iterable, Iterable]:
+        x, inc_list = self.preprocess(x)
+        return x.values.tolist(), inc_list
+    
+    def preprocess_deep_gp(self, x: Iterable) -> Tuple[Iterable, Iterable]:
+        x, inc_list = self.preprocess(x)
+        x_lcs = []
+        for idx in x.index:
+            if idx in self.observations.df.index.levels[0]:
+                budget_level = max(0, self.get_budget_level(x[idx]) - 1)
+                lc = self.observations.extract_learning_curve(
+                    idx, budget_level
+                )
```

Comment:
```suggestion
                budget_level = self.get_budget_level(x[idx])
                lc = self.observations.extract_learning_curve(
                    idx, budget_level
                )
```
Since `- 1` will be moved into the `extract_learning_curve` function